Vice.js

```diff
@@ -0,0 +1,365 @@
+	"lastUpdated": "2017-07-10 08:03:12"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2017 czar
+	http://en.wikipedia.org/wiki/User_talk:Czar
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+// attr()/text()
+function attr(doc,selector,attr,index){if(index>0){var elem=doc.querySelectorAll(selector).item(index);return elem?elem.getAttribute(attr):null}var elem=doc.querySelector(selector);return elem?elem.getAttribute(attr):null}function text(doc,selector,index){if(index>0){var elem=doc.querySelectorAll(selector).item(index);return elem?elem.textContent:null}var elem=doc.querySelector(selector);return elem?elem.textContent:null}
+
+function detectWeb(doc, url) {
+	if (/\/(article|story)\//.test(url)) {
+		return "blogPost";
+	} else if (/\.vice\.com\/?($|\w\w(\_\w\w)?\/?$)|\/(search\?q=)|topic\/|category\/|(latest|read)($|\?page=)/.test(url) && getSearchResults(doc, true) ) {
+		return "multiple";
+	} else if (/amuse-i-d\.vice\.com\/[\w\d-]+\/$/.test(url)) { /* Amuse i-D */
+		return "blogPost";
+	}
+}
+
+function scrape(doc, url) {
+	var item = new Zotero.Item("blogPost");
+	item.blogTitle = attr(doc,'meta[property="og:site_name"]','content') || "Vice";
+	item.language = attr(doc,'html','lang');
+	item.url = url;
+	item.title = text(doc,'.id-title') || attr(doc,'meta[property="og:title"]','content')/* i-D */;
+	item.date = attr(doc,'meta[name="datePublished"]','content') || attr(doc,'meta[property="og:article:published_time"]','content')/* i-D */ || attr(doc,'meta[property="article:published_time"]','content')/* Amuse i-D */ || text(doc,'.post__date')/* Vice News */;
+	if (item.date && item.date.indexOf(" on ") != -1) // clean Vice News
+		item.date = ZU.strToISO(item.date.replace(' on ',''));
+	item.abstractNote = attr(doc,'meta[name="description"]','content');
+	// meta keywords has a bunch of junk, so scrape instead:
+	var keywords = doc.querySelectorAll('.topics li.topic');
+	var keywords_i_D = doc.querySelectorAll('meta[property="og:article:tag"]'); // i-D uses og
+	if (keywords.length) {
+		for (var i=0; i<keywords.length; i++) {
+			item.tags.push(keywords[i].textContent)
+		}
+	} else if (keywords_i_D.length) { 
+		for (var i=0; i<keywords_i_D.length; i++) {
+			item.tags.push(keywords_i_D[i].getAttribute('content'))
+		}		
+	}
+	
+	// archives appear to 404 on load?
```

Comment:
Snapshots on every page I checked will succeed, but when later checked, they all 404. Not sure if there's a workaround or if it would be better to leave the snapshots out altogether.